### PR TITLE
Store test temporary directories outside of `/tmp`

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

There's a long comment inline describing the motivation here.

### CODE REVIEW

File: crates/uv/tests/common/mod.rs [hunk 1]
```diff
@@ -168,10 +172,35 @@ impl TestContext {
...
-        let cache_dir = assert_fs::TempDir::new().expect("Failed to create test cache directory");
-        let python_dir = assert_fs::TempDir::new().expect("Failed to create test Python directory");
-        let home_dir = assert_fs::TempDir::new().expect("Failed to create test home directory");
+        // Discover the path to the XDG state directory. We use this, rather than the OS-specific
+        // temporary directory, because on macOS (and Windows on GitHub Actions), they involve
+        // symlinks. (On macOS, the temporary directory is, like `/var/...`, which resolves to
+        // `/private/var/...`.)
+        //
+        // It turns out that, at least on macOS, if we pass a symlink as `current_dir`, it gets
+        // _immediately_ resolved (so that if call `current_dir` in the subsequent command, you get
```
Comment (zanieb) [inline crates/uv/tests/common/mod.rs:181]: "so that if call"?

File: crates/uv/tests/common/mod.rs [hunk 2]
```diff
@@ -168,10 +172,35 @@ impl TestContext {
...
+        // the resolved symlink). This is problematic, as we _don't_ want to resolve symlinks for
+        // user paths.
```
Comment (zanieb) [inline crates/uv/tests/common/mod.rs:183]:
```suggestion
        // user provided paths.
```

File: crates/uv/tests/common/mod.rs [hunk 3]
```diff
@@ -361,6 +391,7 @@ impl TestContext {
             .env("UV_PYTHON_INSTALL_DIR", "")
             .env("UV_TEST_PYTHON_PATH", self.python_path())
             .env("UV_EXCLUDE_NEWER", EXCLUDE_NEWER)
+            .env("UV_CWD", self.temp_dir.path())
```
Comment (zanieb) [inline crates/uv/tests/common/mod.rs:394]: I presume we can drop this now?

File: crates/uv/tests/common/mod.rs [hunk 4]
```diff
@@ -168,10 +172,35 @@ impl TestContext {
...
+        fs_err::create_dir_all(&cache_dir).expect("Failed to create test cache directory");
+
+        let python_dir = ChildPath::new(root.path()).child("python");
+        fs_err::create_dir_all(&python_dir).expect("Failed to create test Python directory");
+
+        let home_dir = ChildPath::new(root.path()).child("home");
+        fs_err::create_dir_all(&home_dir).expect("Failed to create test home directory");
 
         // Canonicalize the temp dir for consistent snapshot behavior
         let canonical_temp_dir = temp_dir.canonicalize().unwrap();
```
Comment (zanieb) [inline crates/uv/tests/common/mod.rs:206]: Can probably drop this hack too?

File: crates/uv/tests/common/mod.rs (hunk 3 above)
Comment (charliermarsh) [reply crates/uv/tests/common/mod.rs:394]: Yes sorry.

File: crates/uv/tests/common/mod.rs (hunk 4 above)
Comment (charliermarsh) [reply crates/uv/tests/common/mod.rs:206]: I think this needs to come in a future PR, since we still canonicalize internally...

Comment (zanieb) [reply crates/uv/tests/common/mod.rs:206]: In theory canonical temporary directory for testing == non-canonical now, right?

File: crates/uv/tests/common/mod.rs [hunk 5]
```diff
@@ -168,10 +172,35 @@ impl TestContext {
...
+        //
+        // It turns out that, at least on macOS, if we pass a symlink as `current_dir`, it gets
+        // _immediately_ resolved (such that if you call `current_dir` in the running `Command`, it
+        // returns resolved symlink). This is problematic, as we _don't_ want to resolve symlinks
+        // for user-provided paths.
+        let bucket = etcetera::base_strategy::choose_base_strategy()
+            .expect("Failed to find base strategy")
+            .data_dir()
+            .join("uv")
+            .join("tests");
```
Comment (zanieb) [inline crates/uv/tests/common/mod.rs:188]: Sorry, one more ask here — can we respect `UV_INTERNAL__TEST_DIR` or something so downstream maintainers can change this if they need to?

**Review by zanieb — APPROVED** [review]

**Pull request merged**